migrations/20171024163252-host-cleanup.js

```diff
@@ -0,0 +1,318 @@
+
+
+      // check that each entry was filled
+      if (!(withoutFromCollective && withoutFromCollective.id && debit && debit.id && credit && credit.id)) {
+        console.log('>>> withoutFromCollective', withoutFromCollective)
+        console.log('>>> debit', debit);
+        console.log('>>> credit', credit);
+        throw new Error('TransactionGroup check failed');
+      }
+
+      // check that each id is different
+      if (withoutFromCollective.id === debit.id || debit.id === credit.id || credit.id === withoutFromCollective.id) {
+        throw new Error('TransactionGroup duplicate ids found');
+      }
+
+      return { withoutFromCollective, debit, credit }
+    }
+
+    // Change all HostCollectiveIds to orgCollective
+    return sequelize.query(`
+      UPDATE "Transactions"
+        SET "HostCollectiveId" = :orgCollectiveId
+      WHERE "HostCollectiveId" = :superCollectiveId
+      `, { 
+        replacements: {
+          orgCollectiveId: orgCollective.id,
+          superCollectiveId: superCollective.id
+        }})
+
+      // Fetch all transactionGroups that have 3 entries
+      .then(() => sequelize.query(`
+        SELECT "TransactionGroup" FROM "Transactions"
+        WHERE "TransactionGroup" IS NOT NULL AND 
+          ("CollectiveId" = :superCollectiveId OR "FromCollectiveId" = :superCollectiveId)
+        GROUP BY "TransactionGroup"
+        HAVING COUNT(*) >= 3 
+        `, { 
+          type: sequelize.QueryTypes.SELECT,
+          replacements: {
+            superCollectiveId: superCollective.id
+          }}))
+      .then(transactionGroups => {
+        console.log('>>> transaction groups found: ', transactionGroups.length);
+        return transactionGroups;
+      })
+      .each(transactionGroup => {
+
+        console.log('>>> Processing', transactionGroup);
+        // fetch all transaction matching that transactionGroup
+        return sequelize.query(`
+          SELECT * FROM "Transactions" 
+          WHERE "TransactionGroup" = :transactionGroup
+          `, {
+            type: sequelize.QueryTypes.SELECT,
+            replacements: {
+              transactionGroup: transactionGroup.TransactionGroup // data returned this way from query
+            }
+          })
+          .then(transactions => {
+            /* For every 3 set of TransactionGroups, 
+              1. remove transaction with no FromCollectiveId
+              2. take the debit and change CollectiveId to orgCollectiveId
+              3. take the credit and change FromCollectiveId to orgCollectiveId
+            */
+            console.log('>>> transactions', transactions);
+            if (transactions.length !== 3) {
+              throw new Error('Found a transaction group of length', transactions.length);
+            } 
+
+            const tSplit = splitTransactionGroup(transactions);
+
+            // I'm sure there is a way to do this in one query... 
```

Comment:
This didn't work?
```sql
UPDATE "Transactions" SET "deletedAt"=:date, "CollectiveId"=:orgCollectiveId, "FromCollectiveId"=:orgCollectiveId
WHERE id=:tId
```

Doesn't matter, no need to optimize performance for those migrations.